pkg/strategy/xgap/strategy.go

```diff
@@ -257,15 +254,69 @@ func (s *Strategy) makeSpread(ctx context.Context, bestBid, bestAsk types.PriceV
 	return err
 }
 
+func adjustPositionOrder(symbol string, tradingMarket types.Market, currentPosition *types.Position, bestBid, bestAsk types.PriceVolume) (orderForms []types.SubmitOrder) {
+	if currentPosition == nil {
+		log.Warn("current position is nil, do nothing")
+		return
+	}
+	inShortPosition := currentPosition.Base.Sign() < 0
```

Comment:
call .GetBase() for the mutex protection

Reply:
Done